[java-client][okhttp-gson] fixes for interceptors in ApiClient.java

lostiniceland opened the pull request:
ApiClient for Okhttp3 must not copy the interceptors when setting the HttpClient.

fixes #3497

jmini commented on samples/client/petstore/java/okhttp-gson/src/test/java/org/openapitools/client/ApiClientTest.java:
```diff
@@ -335,7 +335,7 @@ public void testInterceptorCleanupWithNewClient() {
         assertEquals(1, oldClient.networkInterceptors().size());
 
-        OkHttpClient newClient = new OkHttpClient();
+        OkHttpClient newClient = apiClient.getHttpClient().newBuilder().build();
```
CircleCI reports that this change gets overridden when the samples are updated:

```diff
diff --git a/samples/client/petstore/java/okhttp-gson/src/test/java/org/openapitools/client/ApiClientTest.java b/samples/client/petstore/java/okhttp-gson/src/test/java/org/openapitools/client/ApiClientTest.java
index 3675d8645a..6181e8afa2 100644
--- a/samples/client/petstore/java/okhttp-gson/src/test/java/org/openapitools/client/ApiClientTest.java
+++ b/samples/client/petstore/java/okhttp-gson/src/test/java/org/openapitools/client/ApiClientTest.java
@@ -335,7 +335,7 @@ public class ApiClientTest {
         OkHttpClient oldClient = apiClient.getHttpClient();
         assertEquals(1, oldClient.networkInterceptors().size());
 
-        OkHttpClient newClient = apiClient.getHttpClient().newBuilder().build();
+        OkHttpClient newClient = new OkHttpClient();
         apiClient.setHttpClient(newClient);
         assertEquals(1, apiClient.getHttpClient().networkInterceptors().size());
         apiClient.setHttpClient(newClient);
```

This is because the source for this file is in the `CI/samples.ci` folder. Can you update this file:
https://github.com/OpenAPITools/openapi-generator/blob/6824bd8b2445e54c25f26589ac51eac3de62f0e9/CI/samples.ci/client/petstore/java/test-manual/okhttp-gson/ApiClientTest.java#L338

jmini commented:
@lostiniceland: Thank you a lot for this PR, can you also change the mentioned file?

lostiniceland commented:
Updated as requested by review.

I would argue though that the Tests I've fixed are not very usefull at all. They might have been with the old "logic" but now the tested method is a plain setter, except the check for equality, which could also be removed.
A usefull invariant would be that the passed client must not be null, because the ApiClient cannot function without a httpClient.

lostiniceland commented:
@jmini I've added another commit which contains the mentioned invariant and removed the unnecessary tests

wing328 commented:
Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/OpenAPITools/openapi-generator/graphs/contributors.

Let me know if you need help fixing it.

Ref: https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

lostiniceland commented:
@wing328 thanks for pointing that out. I've added my work-email to Github so it should match now.

wing328 commented:
@lostiniceland thanks for the PR, which has been included in the 4.1.0 release: https://twitter.com/oas_generator/status/1160000504455319553
